### ISSUE TITLE
IngressRouteTCP object not created when the route is a Service of type ExternalName

### ISSUE THREAD

**Issue by mazzy89** [containous/traefik#6390]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I've created an object of `kind: IngressRouteTCP` and configured in such way:

```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRouteTCP
metadata:
  name: external-tcp-service
spec:
  entryPoints:
    - tcp
  routes:
    - match: HostSNI(`my-external-service.domain.com`)
      services:
        - name: my-external-service
          port: 6379
  tls: {}
---
apiVersion: v1
kind: Service
metadata:
  name: external-tcp-service
spec:
  externalName: internaltcpsrv.domain.local
  type: ExternalName

```

### What did you expect to see?

I would expect that the object `external-tcp-service` would be created.

### What did you see instead?

```sh
level=error msg="Cannot create service: service port not found" 
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2.1.3
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Kubernetes 1.15.9

I believe that the reason is of a bug here https://github.com/containous/traefik/blob/master/pkg/provider/kubernetes/crd/kubernetes_tcp.go#L165

There is an immediate lookup of the port in the `Service` and only after there is a check of the type to verify if the `Service` is `type: ExternalName`.

I believe that the logic must be rearranged a bit. I can wire a PR if is needed.

**Comment by jbdoumenjou** [comment]
Hi @mazzy89, 

Could you explain your use case with the ExternalName Service ?

**Comment by mazzy89** [comment]
Sure. The idea here is to expose a resource specifically an Elasticache cluster that is reachable only from the k8s cluster where Traefik runs.

The k8s `Service` would proxy internally this resource while IngressTCP would create an entry to expose it publicly.

**Comment by mazzy89** [comment]
There is anything I can help here? This is blocking us to adopt Traefik v2. I'm open to hearing different approaches

**Comment by jbdoumenjou** [comment]
You can specify a port on the `ExternalName Service` to map with the `IngressRoute`.
However, it could be a good feature to avoid the need of specifying the port on the `ExternalName Service` by using the `IngressRoute` service port.
 
Just a comment on your sample, the name of the service referenced in the `IngressRoute` does not match the `ExternalName Service` name.

WDYT ?

**Comment by mazzy89** [comment]
> You can specify a port on the ExternalName Service to map with the IngressRoute.
However, it could be a good feature to avoid the need of specifying the port on the ExternalName Service by using the IngressRoute service port.

I'm going to try by specifying the port in the `Service` of type `ExternalName` and see if it works. I was not sure if this is possible. But need definitely to try it.

I like your idea to use the port of the `IngressRoute`. Personally, it does fit our use case.

> Just a comment on your sample, the name of the service referenced in the IngressRoute does not match the ExternalName Service name.

Yeah, that is a typo due to a work of removal of real references from the manifests. 🤓

**Comment by mazzy89** [comment]
Just tried here. Added to the `Service` the port having a `Service` looking like this:

```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRouteTCP
metadata:
  name: external-tcp-service
spec:
  entryPoints:
    - tcp
  routes:
    - match: HostSNI(`my-external-service.domain.com`)
      services:
        - name: external-tcp-service
          port: 6379
  tls: {}
---
apiVersion: v1
kind: Service
metadata:
  name: external-tcp-service
spec:
  externalName: internaltcpsrv.domain.local
  type: ExternalName
  ports:
    - port: 6379
```

The result from Traefik logs

```
+ reverse-proxy-traefikv2-767fd98689-xj2xq › reverse-proxy-traefikv2
reverse-proxy-traefikv2-767fd98689-xj2xq reverse-proxy-traefikv2 time="2020-03-03T14:40:50Z" level=error msg="Cannot create service: service not found" serviceName=external-tcp-service namespace=platform servicePort=6379 providerName=kubernetescrd ingress=external-tcp-service
reverse-proxy-traefikv2-767fd98689-xj2xq reverse-proxy-traefikv2 time="2020-03-03T14:40:50Z" level=error msg="Cannot create service: service not found" servicePort=6379 ingress=external-tcp-service serviceName=external-tcp-service  providerName=kubernetescrd namespace=platform
reverse-proxy-traefikv2-767fd98689-xj2xq reverse-proxy-traefikv2 time="2020-03-03T14:40:51Z" level=error msg="the service \"external-tcp-service-9888b30400de35951949@kubernetescrd\" does not exist" entryPointName=redis routerName=platform-external-tcp-service-9888b30400de35951949@kubernetescrd
```

**Comment by mazzy89** [comment]
@jbdoumenjou I was able to replicate the issue and actually the last one that error is just temporary and due to the order that objects are created. Once swapped the order in the manifest, the errors are not reported anymore and the resources look green in the Traefik dashboard. Need to tests the connection though but at least now Traefik does not complain anymore.

It would be nice to have the feature in place. Do you need help here? Will you go ahead or should I give a try?

**Comment by jbdoumenjou** [comment]
Thank you, we are working on it. :smile:

**Comment by mazzy89** [comment]
Thank you for the fast reaction here.

**Comment by traefiker** [comment]
Closed by #6428.
